### PR TITLE
add sdpa mbart

### PR DESCRIPTION
**Pull request by nbroad1881**
Adds MBartSdpaAttention which is useful for donut.

just copy-pasting from modeling_bart.py

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

~Felix~ :( 

@ArthurZucker

### CODE REVIEW

**Comment by ArthurZucker** [comment]
Will review, could you make sure CIs are green first?

**Comment by nbroad1881** [comment]
🟢

File: src/transformers/models/mbart/modeling_mbart.py [hunk 1]
```diff
@@ -842,6 +956,7 @@ def __init__(self, config: MBartConfig, embed_tokens: Optional[nn.Embedding] = N
         )
         self.layers = nn.ModuleList([MBartEncoderLayer(config) for _ in range(config.encoder_layers)])
         self._use_flash_attention_2 = config._attn_implementation == "flash_attention_2"
+        self._use_sdpa = config._attn_implementation == "sdpa"
```
Comment (ArthurZucker) [inline src/transformers/models/mbart/modeling_mbart.py:959]: this is something we want to avoid, let's just test `config._attn_implementation` directly

**Review by ArthurZucker — COMMENTED** [review]
One nit and should be good to go!

File: src/transformers/models/mbart/modeling_mbart.py (hunk 1 above)
Comment (nbroad1881) [reply src/transformers/models/mbart/modeling_mbart.py:959]: I'm not sure I understand what you are saying when you say "test config._attn_implementation directly"

Comment (nbroad1881) [reply src/transformers/models/mbart/modeling_mbart.py:959]: I was copying this: https://github.com/huggingface/transformers/blob/62c60a30181a65e1a3a7f19c3055a240a6a21335/src/transformers/models/bart/modeling_bart.py#L982

Comment (ArthurZucker) [reply src/transformers/models/mbart/modeling_mbart.py:959]: cc our offline conv!

**Comment by nbroad1881** [comment]
@ArthurZucker 

I finally made the changes :smiley:

**Review by ArthurZucker — APPROVED** [review]
LGTM, could you run the slow tests (which should trigger eager vs sdpa equivalence) ! 🤗

**Comment by nbroad1881** [comment]
How do I run slow tests?

**Comment by nbroad1881** [comment]
We did it!

**Pull request merged**